src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -19,674 +19,557 @@
+                                .url("http://download.elasticsearch.org/[organisation]/elasticsearch-[name]/elasticsearch-[name]-[version].zip")
+                                .url("http://download.elasticsearch.org/elasticsearch/[name]/[name]-latest.zip")
+                                .downloadDir(downloadDir)
+                                .listener(listener)
+                                .timeout(timeout)
+                                .userAgent(userAgent)
+                                .elasticsearchVersion(esVersion)
+                                .build()
+                )
+                .addRepository(
+                        urlRepository("central")
+                                .url("https://repo1.maven.org/maven2/[organisation]/[name]/[version]/[name]-[version].zip")
+                                .url("https://repo1.maven.org/maven2/org/elasticsearch/elasticsearch-[name]/[version]/elasticsearch-[name]-[version].zip")
+                                .url("http://search.maven.org/remotecontent?filepath=[organisation]/[name]/[version]/[name]-[version].zip")
+                                .url("http://search.maven.org/remotecontent?filepath=org/elasticsearch/elasticsearch-[name]/[version]/elasticsearch-[name]-[version].zip")
+                                .downloadDir(downloadDir)
+                                .listener(listener)
+                                .timeout(timeout)
+                                .build()
+                )
+                .addRepository(
+                        urlRepository("sonatype")
+                                .url("https://oss.sonatype.org/service/local/repositories/releases/content/[organisation]/[name]/[version]/[name]-[version].zip")
+                                .url("https://oss.sonatype.org/service/local/repositories/releases/content/[organisation]/elasticsearch-[name]/[version]/elasticsearch-[name]-[version].zip")
+                                .downloadDir(downloadDir)
+                                .listener(listener)
+                                .timeout(timeout)
+                                .build()
+                )
+                .addRepository(
+                        urlRepository("github")
+                                .url("https://github.com/[organisation]/[name]/archive/[version].zip")
+                                .url("https://github.com/[organisation]/[name]/archive/master.zip")
+                                .url("https://github.com/[organisation]/elasticsearch-[name]/archive/[version].zip")
+                                .url("https://github.com/[organisation]/elasticsearch-[name]/archive/master.zip")
+                                .downloadDir(downloadDir)
+                                .listener(listener)
+                                .timeout(timeout)
+                                .build()
+                )
```

Comment:
I really like this part. It makes the whole thing so flexible!
BTW, I have a pending issue which could be added within this PR or later. It's about adding SNAPSHOT sonatype repo as well when we want to share SNAPSHOT versions. See #7672.